### ISSUE TITLE
Handle schema merge for column and FK drop automatically

### ISSUE THREAD

**Issue by fulghum** [dolthub/dolt#6393]
A customer hit a schema merge conflict because their table had dropped a column, an index, and a foreign key. This is a case that Dolt should be able to merge automatically, but for some reason, the merge reported it as a schema conflict and did not allow the user to use the new automatic schema resolution tooling. 

Part of the problem here is that we hit the old schema merge guard we had in place before we added support for automatic schema merging. PR https://github.com/dolthub/dolt/pull/6349 removes that guard, but needs a little more testing to make sure all the cases there are covered by the schema conflict resolution code. Once this is removed, the schema merge code _may_ be able to automatically handle this merge without requiring user intervention. 

**Next steps**:
- complete the work in https://github.com/dolthub/dolt/pull/6349 to remove the old schema merge guard
- add a new test for a merge where a column, an index on the column, and an FK for the column are dropped on one side of the merge
- get the test case above to merge automatically without requiring manual schema altering

**Comment by fulghum** [comment]
I tried to repro this today, but wasn't able to yet. The diff we saw earlier, from the customer, showed a table where a column used in an index and in a FK reference was dropped, but that doesn't seem to be enough to repro the same error message the customer was seeing – in all my attempts, that schema change merged smoothly and automatically. 

Next step is seeing if we can get a clone of the database from the customer, then re-running the same merge and catching the problem in the debugger.
